### PR TITLE
Ansible 2.3 feature support for dellos9 and dellos10

### PR DESCRIPTION
**Pull request by skg-net**
##### SUMMARY
Ansible 2.3 feature support for dellos9 and dellos10

##### ISSUE TYPE
- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME
- ansible/modules/network/dellos9/*
- ansible/modules/network/dellos10/*
##### ANSIBLE VERSION
skg@SKG ~/ansible $ ansible --version
ansible 2.3.0.0 (detached HEAD e4494f85b6) last updated 2017/03/21 19:39:49 (GMT -700)
  config file = 
  configured module search path = Default w/o overrides
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
skg@SKG ~/ansible $ 
##### ADDITIONAL INFORMATION
addresses #22704 and #22705 
With the new Ansible 2.3 infra changes, the dellos modules doesn't work (the new infra changes are not backward compatible), so added the below changes support it.
- Added the new terminal plugin for DellOS9  and DellOS10
- Added the new action plugin for DellOS9 and DellOS10
- Modified the modules to work with the new infra.
with that it adds support for DellOS9 and DellOS10 Persistent Connection support.

### CODE REVIEW

**Comment by skg-net** [comment]
@gundalow @privateip, please take a look at it.

File: lib/ansible/modules/network/dellos9/dellos9_config.py [hunk 1]
```diff
@@ -20,38 +20,37 @@
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-ANSIBLE_METADATA = {'status': ['preview'],
-                    'supported_by': 'core',
-                    'version': '1.0'}
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'core'}
```
Comment (gundalow) [inline lib/ansible/modules/network/dellos9/dellos9_config.py]: Is that right, are these core supported?

**Review by gundalow — CHANGES_REQUESTED** [review]
This seem more than bug fixes, if it is then we need to wait for 2.4. For example you've pulled in the Persistent Connection work. 

A number of modules seem to have changed from "Community" to "Core" supported, is that correct?

**Comment by gundalow** [comment]
Could you please update the description to detail the changes & why they are needed, thanks.

File: lib/ansible/modules/network/dellos9/dellos9_config.py (hunk 1 above)
Comment (skg-net) [reply lib/ansible/modules/network/dellos9/dellos9_config.py]: Yes, you are correct it adds support for Persistent Connection work. The initial understanding was, all the modules which doesn't support Persistent Connection work, will be backward compatible and work with Ansible 2.3. What we found out what its not backward compatible. Hence added the support. 

Regarding the community->core, I believe that was the agreement between Dell & Ansible. If that is not the case let me know.

**Comment by skg-net** [comment]
@gundalow Updated the description with the details and why its needed.

**Comment by privateip** [comment]
needs_rebase

File: lib/ansible/modules/network/dellos10/dellos10_command.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 ANSIBLE_METADATA = {'metadata_version': '1.0',
                     'status': ['preview'],
-                    'supported_by': 'community'}
```
Comment (gundalow) [inline lib/ansible/modules/network/dellos10/dellos10_command.py]: Hi,
These should be `community` still

File: lib/ansible/modules/network/dellos9/dellos9_config.py (hunk 1 above)
Comment (gundalow) [reply lib/ansible/modules/network/dellos9/dellos9_config.py]: These should be `community` still, please revert that change

File: lib/ansible/modules/network/dellos10/dellos10_command.py (hunk 2 above)
Comment (gundalow) [reply lib/ansible/modules/network/dellos10/dellos10_command.py]: This needs reverting back to community

File: lib/ansible/modules/network/dellos10/dellos10_facts.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 #
 ANSIBLE_METADATA = {'metadata_version': '1.0',
                     'status': ['preview'],
-                    'supported_by': 'community'}
```
Comment (gundalow) [inline lib/ansible/modules/network/dellos10/dellos10_facts.py]: This needs reverting back to community

File: lib/ansible/modules/network/dellos9/dellos9_command.py [hunk 4]
```diff
@@ -20,10 +20,9 @@
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-ANSIBLE_METADATA = {'metadata_version': '1.0',
-                    'status': ['preview'],
-                    'supported_by': 'community'}
-
+ANSIBLE_METADATA = {'status': ['preview'],
```
Comment (gundalow) [inline lib/ansible/modules/network/dellos9/dellos9_command.py:23]: This needs reverting back to community

File: lib/ansible/modules/network/dellos9/dellos9_config.py [hunk 5]
```diff
@@ -22,7 +22,7 @@
 
 ANSIBLE_METADATA = {'metadata_version': '1.0',
                     'status': ['preview'],
-                    'supported_by': 'community'}
```
Comment (gundalow) [inline lib/ansible/modules/network/dellos9/dellos9_config.py]: This needs reverting back to community

File: lib/ansible/modules/network/dellos9/dellos9_facts.py [hunk 6]
```diff
@@ -22,7 +22,7 @@
 
 ANSIBLE_METADATA = {'metadata_version': '1.0',
                     'status': ['preview'],
-                    'supported_by': 'community'}
```
Comment (gundalow) [inline lib/ansible/modules/network/dellos9/dellos9_facts.py]: This needs reverting back to community

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/dellos10/dellos10_facts.py (hunk 3 above)
Comment (skg-net) [reply lib/ansible/modules/network/dellos10/dellos10_facts.py]: @gundalow fixed.

File: lib/ansible/modules/network/dellos9/dellos9_command.py (hunk 4 above)
Comment (skg-net) [reply lib/ansible/modules/network/dellos9/dellos9_command.py:23]: @gundalow fixed.

File: lib/ansible/modules/network/dellos9/dellos9_config.py (hunk 5 above)
Comment (skg-net) [reply lib/ansible/modules/network/dellos9/dellos9_config.py]: @gundalow fixed.

File: lib/ansible/modules/network/dellos9/dellos9_facts.py (hunk 6 above)
Comment (skg-net) [reply lib/ansible/modules/network/dellos9/dellos9_facts.py]: @gundalow fixed.

**Comment by gundalow** [comment]
1) The needs fixing
```
Command "test/sanity/code-smell/use-compat-six.sh" returned exit status 1.
>>> Standard Output
lib/ansible/plugins/action/dellos9.py:from ansible.compat.six import iteritems
lib/ansible/plugins/action/dellos10.py:from ansible.compat.six import iteritems
```

2) Also this needs fixing
```
test/sanity/pep8/legacy-files.txt:515:1: A201 Remove "lib/ansible/modules/network/dellos10/dellos10_command.py" since it passes the current rule set
test/sanity/pep8/legacy-files.txt:516:1: A201 Remove "lib/ansible/modules/network/dellos10/dellos10_config.py" since it passes the current rule set
test/sanity/pep8/legacy-files.txt:517:1: A201 Remove "lib/ansible/modules/network/dellos10/dellos10_facts.py" since it passes the current rule set
test/sanity/pep8/legacy-files.txt:520:1: A201 Remove "lib/ansible/modules/network/dellos9/dellos9_command.py" since it passes the current rule set
test/sanity/pep8/legacy-files.txt:885:1: A201 Remove "lib/ansible/plugins/action/dellos10_config.py" since it passes the current rule set
test/sanity/pep8/legacy-files.txt:887:1: A201 Remove "lib/ansible/plugins/action/dellos9_config.py" since it passes the current rule set`
```
Thanks for making these files pep8 clean.
Not then are clean please edit `test/sanity/pep8/legacy-files.txt` in this PR and remove the lines as indicated above

3) You can ignore 

The errors about `vmware_resource_pool.py` will be removed when you push commits

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged into `stable-2.3` https://github.com/ansible/ansible/pull/23010
`legacy-files.txt` isn't updated on `stable-2.3` as the files are not clean.

**Pull request merged**